### PR TITLE
GH-35693: [MATLAB] Add `Valid` as a name-value pair on the `arrow.array.Float64Array` constructor

### LINKED ISSUE #35693: [MATLAB] Add `Valid` as a name-value pair on the `arrow.array.Float64Array` constructor
### Describe the enhancement requested

This is a follow up to #35676 and #35598 .

To allow users to clearly specify which values in an input MATLAB array should be treated as null values when constructing an `arrow.array.Array`, it would be helpful to add a `Valid` name-value pair.

This name-value pair would correspond to the public `Valid` property which encodes the validity bitmap (i.e. null values) in an `arrow.array.Array` (proposed in #35598).

Adding this name-value pair would provide clients with a way to unambiguously control null value handling behavior when constructing `arrow.array.Array` objects.

**Example:**

```matlab
>> matlabArray = double([1, -1, -1, 3, 4, 5])'

matlabArray =

     1
    -1
    -1
     4
     5

>> A = arrow.array.Float64Array(matlabArray, Valid=logical([true false, false, true, true])) 

A = 

[
  1,
  null,
  null,
  4,
  5
]

```

### Component(s)

MATLAB

### PR DESCRIPTION
**Pull request by sgilmore10**
### Rationale for this change
Users should be able to specify exactly which elements in a MATLAB array should be treated as `null` values. In a previous PR, we added the name-value pair `InferNulls` which users can use to turn off the automatic "null detection" of values for which the MATLAB `ismissing` function returns true:

```MATLAB
>> matlabArray = [1 NaN 3];

% NaN is treated as null by default
>> arrowArray = arrow.array.Float64Array(matlabArray);

arrowArray = 

[
  1,
  null,
  3
]

% Don't treat NaN as a null value
>> arrowArray = arrow.array.Float64Array(matlabArray, InferNulls=false);

arrowArray = 

[
  1,
  nan,
  3
]
``` 

To provide more control, this PR adds a new name-value pair named `Valid`. This name-value pair will take precedence over `InferNulls` and allows users to specify which elements are valid using either a logical array or a list of indices:

```MATLAB
>> matlabArray = [1 NaN 3];

>> arrowArray = arrow.array.Float64Array(matlabArray, Valid=[true true false]);

arrowArray = 

[
  1,
  nan,
  null
]

>> arrowArray = arrow.array.Float64Array(matlabArray, Valid=[2 3]);

arrowArray = 

[
  null,
  nan,
  3
]

``` 

### What changes are included in this PR?

1. Added the `Valid` name-value pair to `arrow.array.Float64Array`.

### Are these changes tested?

1. Added a new test file called `tParseValidElements.m` to independently test the helper function `arrow.args.parseValidElements` used by `arrow.array.Float64Array`.
2. Added two integration tests in `tFloat64Array.m`.

### Are there any user-facing changes?
Yes, users can now pass the `Valid` name-value pair to `arrow.array.Float64Array`.

### Future Directions

1. Currently, only `arrow.array.Float64Array` supports the `InferNulls` and `Valid` name-value pairs. In a followup PR, we will add support for these name-value pairs to the rest of the numeric array classes. 

* Closes: #35693

### CODE REVIEW

File: matlab/test/arrow/args/tParseValidElements.m [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+% permissions and limitations under the License.
+
+classdef tParseValidElements < matlab.unittest.TestCase
+    % Tests for arrow.args.parseValidElements
+    
+    methods(Test)
+        % Test methods
+        function InferNullsTrue(testCase)
+        % Verify values for which ismissing returns true for are not
+        % considered valid.
```
Comment (kou) [inline matlab/test/arrow/args/tParseValidElements.m:23]:
```suggestion
            % Verify values for which ismissing returns true for are not
            % considered valid.
```

File: matlab/test/arrow/args/tParseValidElements.m [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+            data = [1 2 3];
+            validElements = parseValidElements(data, InferNulls=true);
+            expectedValidElements = [true; true; true];
+            testCase.verifyEqual(validElements, expectedValidElements);
+        end
+
+        function InferNullsFalse(testCase)
+        % Verify all elements are treated as Valid wen InferNulls=false is
+        % provided - including values for which that ismissing returns true.
```
Comment (kou) [inline matlab/test/arrow/args/tParseValidElements.m:37]:
```suggestion
            % Verify all elements are treated as Valid wen InferNulls=false is
            % provided - including values for which that ismissing returns true.
```

File: matlab/test/arrow/args/tParseValidElements.m [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+            % Supply a numeric vector for Valid 
+            validElements = parseValidElements(data, Valid=[1 3]);
+            expectedValidElements = [true; false; true];
+            testCase.verifyEqual(validElements, expectedValidElements);
+        end
+
+        function ComplexValidIndicesError(testCase)
+        % Verify parseValidElements errors if Valid is a numeric array
+        % containing complex numbers.
```
Comment (kou) [inline matlab/test/arrow/args/tParseValidElements.m:79]:
```suggestion
            % Verify parseValidElements errors if Valid is a numeric array
            % containing complex numbers.
```

Is this natural comment indent style in MATLAB...?

**Review by kou — APPROVED** [review]
+1

File: matlab/test/arrow/args/tParseValidElements.m (hunk 3 above)
Comment (sgilmore10) [reply matlab/test/arrow/args/tParseValidElements.m:79]: Either comment style works well. Typically, comments that describe the overall behavior of a function are not indented. These comments become the help text that appears when you run: 

```MATLAB
>> help <functionName>
```
Here's a [link](https://www.mathworks.com/help/matlab/matlab_prog/add-help-for-your-program.html) to a documentation page about help text if you are interested in reading more about it. However, it's fine for these comments to be indented, so it's really just a personal preference.

Comment (kou) [reply matlab/test/arrow/args/tParseValidElements.m]: > Typically, comments that describe the overall behavior of a function are not indented.

Oh, I see. Then could you revert my suggestions? Sorry.

Comment (sgilmore10) [reply matlab/test/arrow/args/tParseValidElements.m]: I think it's ok if we leave the comments indented. Both styles are acceptable in MATLAB. I talked with @kevingurney about this yesterday and he prefers indenting these comments. I don't have a strong opinion, so I'm happy to go with indenting them. Does that work for you @kou?

Comment (kou) [reply matlab/test/arrow/args/tParseValidElements.m]: OK!
I'll merge this after you push the changes.

Comment (sgilmore10) [reply matlab/test/arrow/args/tParseValidElements.m]: Thanks Kou for working with us!

**Pull request merged**